cpu/registers.go

```diff
@@ -0,0 +1,75 @@
+package cpu
+
+// Register represents a typical 16-bit register.
+type Register uint16
+
+// Hi returns the most-significant 8 bits in a register.
+// For example, in the bit string representation of a register: 1111_1111_0000_0000
+// the first eight "1" bits would be returned as a uint8.
+func (r Register) Hi() uint8 {
+	return uint8(r >> 8)
+}
+
+// SetHi sets the most-significant 8 bits in the register.
+func (r *Register) SetHi(x uint8) {
+	*r |= Register(x) << 8
+}
+
+// Lo returns the least-significant 8 bits in a register.
+// For example, in the bit string representation of a register: 0000_0000_1111_1111
+// the trailing eight "1" bits would be returned as a uint8
+func (r Register) Lo() uint8 {
+	return uint8(r)
+}
+
+// SetLo sets the least-significant 8 bits in the register
+func (r *Register) SetLo(x uint8) {
+	*r |= Register(x)
+}
+
+// Registers represents the entire Register file of a Gameboy CPU.
+type Registers struct {
+	// The high bits of AF is called the "accumulator", and the low bits are the flags.
+	AF Register
+	BC Register
+	DE Register
+	HL Register
+
+	SP Register
+	PC Register
+}
+
+// Flags returns the Flags register from the Lo bits of the AF register.
+func (r *Registers) Flags() Flags {
+	return Flags(r.AF.Lo())
+}
+
+// Flags represents an 8-bit flags register, found in the lower 8-bits of the AF register.
+type Flags uint8
```

Comment:
Not really, Flags is special because  its 8 bits decompose into special individual values with distinct, and important meanings. The upper 8 bits representing `A` from `AF` are just any 8 bits used for anything.